web/src/components/RosterUpdateForm.tsx

```
@@ -59,525 +47,487 @@ interface RosterApiItem extends Omit<RosterWithChampion, 'champion'> {
+    const [rank, setRank] = useState("3");
+    const [isAscended, setIsAscended] = useState(false);
+    const [files, setFiles] = useState<File[]>([]);
+    const [previews, setPreviews] = useState<string[]>([]);
+
+    const [uploadProgress, setUploadProgress] = useState(0);
+    const [statusMessage, setStatusMessage] = useState("");
+
+    // Cleanup object URLs to avoid memory leaks
+    useEffect(() => {
+        return () => {
+            previews.forEach(url => URL.revokeObjectURL(url));
```

Biome (2.4.4)

[error] This callback passed to forEach() iterable method should not return a value.

(lint/suspicious/useIterableCallbackReturn)